### PR TITLE
First pull request. Sorry Chris!

### PR DESCRIPTION
**Pull request by hBar2013**

### CODE REVIEW

**Review by hughjafro — COMMENTED** [review]
**Module 1:**
- Start a clean notebook.
  - Make sure you use comments in your code using # or text cells
- Do train/validate/test split with the Tanzania Waterpumps data.
  - Explore different train_size and test_size
- Begin to explore and clean the data.
  - Import has the correct shape
  - Good choice to evaluate value counts to further explore data size
  - Using groupby to analyze data was a good choice for EDA. Try to add comments about what your output is telling you about the data.
  - Effective way to address zeroes and NaNs and methodical approach to analyze each feature
  - Ingenious to use .now for aging purposes
- Select different numeric and categorical features.
  - The Permit feature has quite a few missing values
- Do one-hot encoding. (Remember it may not work with high cardinality categoricals.)
- Scale features.
- Use scikit-learn for logistic regression.
  - Check
- Get your validation accuracy score.
  - nice score
- Get and plot your coefficients.
  - Make sure to plot the coeff with something like
    - coefficients = pd.Series(model.coef_[0], X_train_encoded.columns)
    - plt.figure(figsize=(10, 10))
    - coefficients.sort_values().plot.barh();
- Submit your predictions to our Kaggle competition.
- Commit your notebook to your fork of the GitHub repo.

**Module2 Decision Tree:**
- Good implementation of decision tree
- Hopefully, you are exploring various features
- Stretch: How does changing the train/test size affect your score?

**Review by hughjafro — COMMENTED** [review]
### Module 03 Random Forest:

**Overall:** Great job getting into the 80's.
**Feedback:** Comment your code. Try more feature engineering.

_Do more exploratory data analysis, data cleaning, feature engineering, and feature selection._
- Changing the features that are used and even creating new features may give you a performance boost. Would squaring certain features be justified?

_Try a Random Forest Classifier._
- Check

_Try Ordinal Encoding._
- You may have done this in a later version, but this is worth attempting.

**Submit new predictions.**

**Comment by hughjafro** [comment]
### Module 04 Confusion Matrix:

**Overall:** Great job this week, despite the :electric_plug: issues.
**Feedback:** COMMENT YOUR CODE. Future Kim will appreciate it...a lot.

_Get and interpret the confusion matrix for classification models_
- Nice function for model selection
- Your confusion matrix plot is well done
- What does the c-matrix tell you?
- What is the significance of the precision, recall, f-1 scores?

_Kaggle Challenge_
- Ask yourself, what you would do differently on your next Kaggle challenge? In approach and execution.

**Great Job!!!**
